server/src/main/java/org/elasticsearch/common/bytes/AbstractBytesReference.java

```diff
@@ -26,6 +26,11 @@ public int getInt(int index) {
         return (get(index) & 0xFF) << 24 | (get(index + 1) & 0xFF) << 16 | (get(index + 2) & 0xFF) << 8 | get(index + 3) & 0xFF;
     }
 
+    @Override
+    public int getIntLE(int index) {
```

Comment:
We should add an implementation (using var handles as we did for other stuff in `Numbers`) of this to `BytesArray`, otherwise this might be quite slow if it's not getting inlined properly (which it probably won't be).

Reply:
Yeah. We have stuff in `ByteUtils` for this sort of thing I think. I figured I'd do it in a followup and add a microbenchmark. I could do it now if you'd prefer too.

Reply:
Follow-up seems fine by me, especially if it gets a benchmark, thanks! :)